yaptide/routes/common_sim_routes.py

```diff
@@ -168,12 +172,21 @@ def post():
         if decoded_token != sim_id:
             return yaptide_response(message="Invalid update key", code=400)
 
-        for estimator_dict in payload_dict["estimators"]:
+        for i, estimator_dict in enumerate(payload_dict["estimators"]):
             # We forsee the possibility of the estimator being created earlier as element of partial results
             estimator = fetch_estimator_by_sim_id_and_est_name(sim_id=sim_id, est_name=estimator_dict["name"])
-
             if not estimator:
-                estimator = EstimatorModel(name=estimator_dict["name"], simulation_id=simulation.id)
+                file_name = estimator_dict["name"]
+                if simulation.sim_type == SimulationType.FLUKA.value and len(simulation.inputs) > 0:
+                    # Fluka preserves the order of estimators in the input data.
+                    # We can use estimator name from the input data
+                    estimator_name = simulation.inputs[0].data["input_json"]["scoringManager"]["outputs"][i]["name"]
+                else:
+                    # ShieldHit does not preserve the order,
+                    # but the estimator_name is equal to file_name without '_' at the end
+                    estimator_name = file_name[:-1] if file_name[-1] == "_" else file_name
```

Comment:
wouldn't be simpler to fix that bug which introduces `_` at the end of the name instead of adding in another place such a code ?

Reply:
So if we have the case of `simulation.sim_type == SimulationType.FLUKA.value and len(simulation.inputs) == 0:` you would check here for Fluka output name by doing a check for last letter to be `_` ?

Reply:
Here is a source of all trailing underscore issues:

https://github.com/DataMedSci/pymchelper/blob/e862c6ef638fa9470286c7154fa2ef9b782cc1c2/pymchelper/readers/shieldhit/reader_base.py#L29-L46

```python
[ares][plgkongruencj@ac0788 simple]$ python3
Python 3.9.6 (default, Oct  6 2021, 13:38:06)
[GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> basename = 'ex_yzmsh_0001.bdo'
>>> core_name = basename[:-4]
>>> basename[-8:-4].isdigit() and len(basename[-8:-4]) == 4
True
>>> core_name = basename[:-8]
>>> core_name
'ex_yzmsh_'
```

Example SHIELD-HIT12A output files:
```
[ares][plgkongruencj@ac0788 simple]$ ll
total 329
-rw-r--r-- 1 plgkongruencj plgrid    512 Dec  6 21:00 beam.dat
-rw-r--r-- 1 plgkongruencj plgrid   1040 Dec  6 21:00 detect.dat
-rw-r--r-- 1 plgkongruencj plgrid   1278 Dec  6 21:00 detect_old.dat
-rw-r--r-- 1 plgkongruencj plgrid  26717 Dec  7 21:18 ex_cyl_0001.dat
-rw-r--r-- 1 plgkongruencj plgrid  26717 Dec  7 21:18 ex_cyl.dat
-rw-r--r-- 1 plgkongruencj plgrid 722520 Dec  7 21:18 ex_yzmsh_0001.bdo
-rw-r--r-- 1 plgkongruencj plgrid 722512 Dec  7 21:18 ex_yzmsh.bdo
-rw-r--r-- 1 plgkongruencj plgrid    606 Dec  6 21:00 geo.dat
-rw-r--r-- 1 plgkongruencj plgrid     22 Dec  6 21:00 mat.dat
-rw-r--r-- 1 plgkongruencj plgrid   1536 Dec  6 21:00 README.md
-rw-r--r-- 1 plgkongruencj plgrid 131529 Dec  7 21:18 shieldhit_0001.log
-rw-r--r-- 1 plgkongruencj plgrid 131510 Dec  7 21:18 shieldhit.log
```

Reply:
https://github.com/DataMedSci/pymchelper/pull/769 proposition of a fix

Reply:
For the purpose of testing you can put in yaptide backend following line in requirements list.

```
[tool.poetry.dependencies]
pymchelper = { git = "https://github.com/DataMedSci/pymchelper.git", branch = "768-fix-trailing-underscore-_-issue-in-the-file-corename" }
```

If all the tests are OK, then I can merge the PR and release new version of pymchelper.